### ISSUE TITLE
exit status 1: 'C:\Users\Konrad' is not recognized as an internal or external command, operable program or batch file.

### ISSUE THREAD

**Issue by KonradLinkowski** [coreybutler/nvm-windows#405]
### My Environment

- [x] Windows 10
- [x] My Windows installation is non-English.

### I'm using NVM4W version:

- [x] 1.1.7

### I have already...

- [x] read the README to be aware of npm gotchas & antivirus issues.
- [ ] reviewed the [wiki](https://github.com/coreybutler/nvm-windows/wiki) to make sure my issue hasn't already been resolved.
- [ ] verified I'm using an account with administrative privileges.
- [ ] searched the issues (open and closed) to make sure this isn't a duplicate.
- [ ] made sure this isn't a question about how to use NVM for Windows, since [gitter](https://gitter.im/coreybutler/nvm-windows?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge) is used for questions and comments.

### My issue is related to (check only those which apply):

- [ ] settings.txt
- [ ] proxy support (Have you tried version 1.1.0+?)
- [ ] 32 or 64 bit support (Have you tried version 1.1.3+?)
- [ ] Character escaping (Have you tried version 1.1.6+?)
- [x] A standard shell environment (terminal/powershell)
- [ ] A non-standard shell environment (Cmder, Hyper, Cygwin, git)

### Expected Behavior

Change node version.

### Actual Behavior

exit status 1: 'C:\Users\Konrad' is not recognized as an internal or external command,
operable program or batch file.

It spited my user folder in half by space.

### Steps to reproduce the problem:

nvm use 11.1.0

**Comment by Mourdraug** [comment]
yep, seems like your user folder cannot contain whitespaces.

**Comment by marcoancona** [comment]
I see the same problem

**Comment by Mourdraug** [comment]
Oh I was going to write this earlier but I forgot. You can actually edit config file to use DOS path convention (you can find correct short directory name with `dir /x` command) and theoretically easiest fix would be to create config using this convention in the first place, but this sounds more like hiding a bug than fixing it.

**Comment by KonradLinkowski** [comment]
I would expect nvw-windows to works on windows by default.

**Comment by algoshipda** [comment]
same problem +1

**Comment by alexkar13** [comment]
thanks @Mourdraug it worked !

Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA\~1 in my case) 
Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

**Comment by coreybutler** [comment]
This is fixed in master. I just haven't had time to cut a release of 1.1.8.

**Comment by AtomicLemon** [comment]
If you used scoop to install nvm, the directory for the settings.txt is

**Comment by AtomicLemon** [comment]
If you used scoop to install nvm, the directory for the settings.txt file is

C:\Users\John Doe\scoop\persist\nvm\settings.txt

**Comment by stefanbschneider** [comment]
Intsalled in `C:/Program Files/..` couldn't find a workaround to get it working so far...

**Comment by Integrion** [comment]
Hi, i just encountered the same issue on a Windows 10 system... i suspected that the user root path string parser can't deal with a space anywhere in the root path name... so i gave it the 7.3  DOS user path name instead as suggested above... J52B0~1.CLA in our case, and that indeed worked just fine, thanks.

**Comment by apihlaja** [comment]
Yep, that's still issue, 1.1.8 isn't out yet ie. the proper fix is still laying somewhere in commit log waiting for the next release. Until then, workaround mentioned above should do the job.

**Comment by stevebaxter** [comment]
Still broken - spaces in file paths have been a thing for quite a while now!

**Comment by AbhiSinghDeveloper** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case)
> Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
> root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

C:\Users\ ABHISH~1\AppData\Roaming\nvm could not be found or does not exist. Exiting.
node v11.0.0 (64-bit) is not installed.
It's now showing this error

**Comment by savjisuyash-code** [comment]
1] In cmd go to the home directory (User in my case)  Run "dir /x" 
2] Find the short name for your directory (In my case Suyash Savji = SAVJIS~1)
3] Go to C:\Users\"Home File"\AppData\Roaming\nvm
4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~1 )
5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

**Comment by savjisuyash-code** [comment]
1] In cmd go to the home directory (User in my case)  Run "dir /x" 
2] Find the short name for your directory (In my case Suyash Savji = SAVJIS~1)
3] Go to "C:\Users\"Home File"\AppData\Roaming\nvm"
4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~1 )
5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

**Comment by savjisuyash-code** [comment]
1] In cmd go to the home directory (User in my case)  Run "dir /x" 
2] Find the short name for your directory (In my case Suyash Savji = SAVJIS ~~ 1)
3] Go to C:\Users\"Home File"\AppData\Roaming\nvm
4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~~1 )
5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

**Comment by laurentmenard** [comment]
Great ! Follox the 5 steps, and it's working

**Comment by PratikSaha198** [comment]
Real big thank you for the solution !

**Comment by ukaemma2** [comment]
thank you so much this is really helpful

**Comment by BHarshavardhanReddy** [comment]
> 1] In cmd go to the home directory (User in my case) Run "dir /x"
> 2] Find the short name for your directory (In my case Suyash Savji = SAVJIS ~~ 1)
> 3] Go to C:\Users"Home File"\AppData\Roaming\nvm
> 4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~~1 )
> 5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

where to find short name of our directory

**Comment by Swarag-N** [comment]
### tl;dr
### Solved!!
For someone who is not able to work with ~1 method by @kirkouimet  (Thanks mate) in  #41 
Open CMD go to the Users folder and Get the Shortened Folder name.

and type 

### > `dir\x `

![Annotation 2020-05-25 112714](https://user-images.githubusercontent.com/46181010/82782765-c2d29200-9e7a-11ea-83ed-6173ddd85948.png)

Now later after copying or Noting down the folder name
![Annotation 2020-05-25 113201](https://user-images.githubusercontent.com/46181010/82783088-69b72e00-9e7b-11ea-8715-e721d7961c8c.png)

Hope it helps

**Comment by KonradLinkowski** [comment]
I have to say that fixing this problem manually by the user isn't a fix at all for me.
Windows can have spaces in folders names and many users have this (because of microsoft account) by default.
The fix should be provided in the nvm itself by developers not by the users who want to use it on their end.

**Comment by shahkv95** [comment]
@savjisuyash-code Thank you so much. it worked...!

**Comment by cfdeveloper84** [comment]
@Swarag-N , thanks for the screenshots

**Comment by cornel37** [comment]
Thank you so much for this

**Comment by sgoyal2402** [comment]
I DONT have any shortened user name what should i do??
![Capture](https://user-images.githubusercontent.com/64013350/84260673-0fcf8d00-ab38-11ea-9dc3-73fe8b259192.PNG)

**Comment by aaztha** [comment]
> I DONT have any shortened user name what should i do??
> ![Capture](https://user-images.githubusercontent.com/64013350/84260673-0fcf8d00-ab38-11ea-9dc3-73fe8b259192.PNG)

**Comment by aaztha** [comment]
> I DONT have any shortened user name what should i do??
> ![Capture](https://user-images.githubusercontent.com/64013350/84260673-0fcf8d00-ab38-11ea-9dc3-73fe8b259192.PNG)

i am also facing the same problem

**Comment by aaztha** [comment]
It didn't work for me...didn't get any shortened user name

**Comment by Heet1996** [comment]
It didn't work for me because I am not getting shortened name for my user

**Comment by Ashanmg** [comment]
Faced the same issue and resolved it using these help

**Comment by junnysmiles** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case)
> Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
> root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

this worked for me. thank you so much!

**Comment by clevengr** [comment]
> This is fixed in master. I just haven't had time to cut a release of 1.1.8.

So coming up on 2 years and the download installer is still 1.1.7...  I get that this is free software (and I'm sure it will be VERY USEFUL once I get it working), but just out of curiosity, how long IS your development cycle?  ;)

**Comment by leonardoraele** [comment]
Still broken as of Dec/2020

**Comment by albert2k21** [comment]
> Hi, i just encountered the same issue on a Windows 10 system... i suspected that the user root path string parser can't deal with a space anywhere in the root path name... so i gave it the 7.3 DOS user path name instead as suggested above... J52B0~1.CLA in our case, and that indeed worked just fine, thanks.

How did you do this?

**Comment by albert2k21** [comment]
> > thanks @Mourdraug it worked !
> > Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case)
> > Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
> > root: C:\Users\ALEXKA~1\AppData\Roaming\nvm
> 
> this worked for me. thank you so much! Especially given that doing everything else would take more time. this is awesomeeeeee

How did you do this?

**Comment by ashna99** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case)
> Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
> root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

Thanks@Mourdraug it really solved the issue

**Comment by VirtueAndTerror** [comment]
> 
> 
> 1] In cmd go to the home directory (User in my case) Run "dir /x"
> 2] Find the short name for your directory (In my case Suyash Savji = SAVJIS ~~ 1)
> 3] Go to C:\Users"Home File"\AppData\Roaming\nvm
> 4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~~1 )
> 5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

I followed the 5 steps and solve the issues. Thank you!

**Comment by onyet** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case)
> Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
> root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

Thanks... you save my time. 🥇

**Comment by shivam67847** [comment]
> 1] In cmd go to the home directory (User in my case) Run "dir /x"
> 2] Find the short name for your directory (In my case Suyash Savji = SAVJIS ~~ 1)
> 3] Go to C:\Users"Home File"\AppData\Roaming\nvm
> 4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~~1 )
> 5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

worked for me, thanks ....
. you saved my time

**Comment by shopifyfullstackdev** [comment]
> I DONT have any shortened user name what should i do??
> ![Capture](https://user-images.githubusercontent.com/64013350/84260673-0fcf8d00-ab38-11ea-9dc3-73fe8b259192.PNG)



> ### tl;dr
> ### Solved!!
> For someone who is not able to work with ~1 method by @kirkouimet (Thanks mate) in #41
> Open CMD go to the Users folder and Get the Shortened Folder name.
> 
> and type
> 
> ### > `dir\x `
> ![Annotation 2020-05-25 112714](https://user-images.githubusercontent.com/46181010/82782765-c2d29200-9e7a-11ea-83ed-6173ddd85948.png)
> 
> Now later after copying or Noting down the folder name
> ![Annotation 2020-05-25 113201](https://user-images.githubusercontent.com/46181010/82783088-69b72e00-9e7b-11ea-8715-e721d7961c8c.png)
> 
> Hope it helps

Thank you so much

**Comment by Fighteros** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case)
> Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this :
> root: C:\Users\ALEXKA~1\AppData\Roaming\nvm


thanks is fixed my problem

**Comment by coreybutler** [comment]
Spaces are now supported in https://github.com/coreybutler/nvm-windows/releases/tag/1.1.8

**Comment by KonradLinkowski** [comment]
Finally 🌟 , thank you very much 😄

**Comment by SURUCHIWALIA12** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case) Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this : root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

worked well for me.

**Comment by Arjuncse07** [comment]
Facing the same issue

**Comment by vaishalparikh** [comment]
> @savjisuyash-code  Real big thank you for the solution !

**Comment by qyaaaa** [comment]
Still broken as of Mar/2023.

**Comment by qyaaaa** [comment]
Still broken as of Mar/2023.

**Comment by davec-fvr** [comment]
Still having this issue in 1.1.10

**Comment by tyhuynh03** [comment]
Thank it really solved the issue!!

**Comment by bhati00** [comment]
> thanks @Mourdraug it worked !
> 
> Open Cmd, go to the Users directory, type dir /x and you will see the shortened version of the name of the users. (ALEXKA~1 in my case) Inside the nvm installation (AppData\Roaming\nmv in my case) there is a settings file, edit the first line and instead of having the space name put that shortened version and leave the rest the same like this : root: C:\Users\ALEXKA~1\AppData\Roaming\nvm

@alexkar13 , thx man , you saved my day .

**Comment by Sciec** [comment]
Thanks , this helped me achieving it.

**Comment by galahzhenya** [comment]
Still have issue in nvm v1.1.10.

**Comment by waseber** [comment]
Important information for the short name solution: `dir /x`  does not work in Power Shell, it NEEDS to be in the simple command prompt.

**Comment by kriyanshishah** [comment]
> 1] In cmd go to the home directory (User in my case) Run "dir /x" 2] Find the short name for your directory (In my case Suyash Savji = SAVJIS ~~ 1) 3] Go to C:\Users"Home File"\AppData\Roaming\nvm 4] Open settings.txt file and replace the home file name with the short name(In my case, Suaysh Savji :: SAVJIS~~1 ) 5] In cmd go to the version file installed eg. v11.0.0 and then run "nvm use 11.0.0"

This worked.
thanks

**Comment by GetNameSniped** [comment]
I am still facing this issue in nvm 1.2.2

**Comment by jo-bert** [comment]
It was simple fix. Remove old nvm and then install the nvm in C:\nvm path.
